### PR TITLE
Unify bi-operators throughout the library #2190 

### PR DESCRIPTION
**Pull request by thadumi**
Added missing methods to Try, Option, Either, and Validation according to issue #2190

### CODE REVIEW

File: src/main/java/io/vavr/control/Try.java [hunk 1]
```diff
@@ -694,6 +694,26 @@ public final boolean isSingleValued() {
...
+     * @param mapSuccess A mapping function taking the current value
+     * @param <U> The new component type
+     * @return
+     */
+    public final <U> Try<U> map(Function<? super Throwable, ? extends U> mapFailure, Function<? super T, ? extends U> mapSuccess) {
+        Objects.requireNonNull(mapFailure, "mapFailure is null");
+        Objects.requireNonNull(mapSuccess, "mapSuccess is null");
+
+        if (isFailure()) {
+            return Try.of(() -> mapFailure.apply(getCause()));
```
Comment (thadumi) [inline src/main/java/io/vavr/control/Try.java]: Map with both parameters for a Try could have two behaviours. In the first one the mapFailure parameter acts as a recover function, while in the second one mapFailure acts as a mapping from Throwable to ? extends Throwable. 

In the second case, the behaviour will be the same as calling `toEither().map(mapFailure, mapSuccess)`, and, at the same time, we can ensure that the left contains only a Throwable value. Instead in the first scenario calling `toEither().map(mapFailure, mapSuccess)` feels like having an Either<Recover_Type, Success_Type>.

What do you think about it?

Comment (danieldietrich) [reply src/main/java/io/vavr/control/Try.java]: Hi Theodor, that‘s a good question. I‘m currently sitting in a meetup and will take a look later. But I see parallels to the existing fold() and transform() methods. Maybe the map() methods are kind of redundant here...

Comment (danieldietrich) [reply src/main/java/io/vavr/control/Try.java]: @thadumi I've double-checked the 2.0.0 branch, where I already hardened the upcoming API.

Several methods will change, especially [`transform`](https://github.com/vavr-io/vavr/blob/v2.0.0/src/main/java/io/vavr/control/Try.java#L701-L711):

```java
// Vavr 1.0
<U> U transform(Function<Try<T>, U> f)

// Vavr 2.0
<U> Try<U> transform(CheckedFunction<Throwable, Try<U>> ifFailure, CheckedFunction<T, Try<U>> ifSuccess)
```

`fold` is already there and will stay that way:

```java
<U> U fold(Function<Throwable, U> ifFailure, Function<T, U> ifSuccess)
```

By using `fold` and `transform` we are already able to map the underlying values:

```java
transform(err => Failure(new Error("")), x => Success("whatever"))
```

When Java Records & Switch Expressions & Pattern Matching & Object Deconstruction will be available, a convenient way in Java would be:

```java
var mappedTry = switch(myTry) {
    case Success(val) -> mapSuccess.apply(val)
    case Failure(err) -> mapFailure.apply(err)
}
```

I know that these solutions are a bit more verbose than the map overload we thought about. But it is important to me that we keep the API surface area small. There are already several options to achieve a similar result.

So let's keep the new `peek` and abandon the `map` overloads. Okay?

Comment (thadumi) [reply src/main/java/io/vavr/control/Try.java]: Yes, I see your point. 
I just removed them.

File: src/main/java/io/vavr/control/Either.java [hunk 2]
```diff
@@ -795,6 +816,10 @@ private LeftProjection(Either<L, R> either) {
             return either.<L2, R2> bimap(leftMapper, rightMapper).left();
         }
 
+        public <L2, R2> LeftProjection<L2, R2> map(Function<? super L, ? extends L2> leftMapper, Function<? super R, ? extends R2> rightMapper) {
+            return bimap(leftMapper, rightMapper);
+        }
```
Comment (danieldietrich) [inline src/main/java/io/vavr/control/Either.java]: Please remove this map method

File: src/main/java/io/vavr/control/Either.java [hunk 3]
```diff
@@ -1050,6 +1075,10 @@ private RightProjection(Either<L, R> either) {
             return either.<L2, R2> bimap(leftMapper, rightMapper).right();
         }
 
+        public <L2, R2> RightProjection<L2, R2> map(Function<? super L, ? extends L2> leftMapper, Function<? super R, ? extends R2> rightMapper) {
+            return bimap(leftMapper, rightMapper);
+        }
```
Comment (danieldietrich) [inline src/main/java/io/vavr/control/Either.java]: Please remove this map method

File: src/main/java/io/vavr/control/Validation.java [hunk 4]
```diff
@@ -777,8 +777,30 @@ public final void forEach(Consumer<? super T> action) {
...
+        } else {
+            validAction.accept(get());
+        }
+
+        return this;
+    }
+
     @Override
     public final Validation<E, T> peek(Consumer<? super T> action) {
+        Objects.requireNonNull(action, "action is null");
```
Comment (danieldietrich) [inline src/main/java/io/vavr/control/Validation.java]: Great!

**Pull request merged**